lib/extractor/callbacks.ts

```diff
@@ -20,13 +18,17 @@ export async function applyCallbacks(
         : // If no callback was provided for this action then return as string by default.
           streamToString(streamCopy);
 
-    return promise.then((content) => {
-      // Assign the result once the Promise is complete.
-      result[action.actionName] = content;
-    });
-  });
+    const content = await promise;
```

Comment:
I think this was intentionally made to fire all promises at the same time and we should bring it back to creating all promises and then awaiting all. If you await them one by one the stream duplication and processing cannot work because by the time the 2nd action is run, the stream is already drained/read.

Reply:
Thanks a lot for catching, updating 👌